src/components/Button.js

```diff
@@ -1,16 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-export default function Button({ name }) {
+export default function Button({ name, color, width }) {
+  let buttonClass = 'button';
+
+  if (color) {
+    buttonClass += ` ${color}`;
+  }
+
+  if (width) {
+    buttonClass += ` ${width}`;
+  }
+
   return (
-    <button type="button" className="button">{name}</button>
+    <button type="button" className={buttonClass}>{name}</button>
   );
 }
 
 Button.defaultProps = {
   name: '',
+  color: '',
```

Comment:
the color prop has a default of orange ![see this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png)

Reply:
as I understand, the `color` prop doesn't need a default value, the button should be orange since the begging as the image says `the button should have an orange background by default`. So, it is orange, and if the prop is passed it changes the color to gray.

Reply:
@oscardelalanza I think you are both trying to achieve the same thing. However, using it here makes it clear what the default colour is.

Another bright side is that you won't need to check if `color` is valid before adding it to the `buttonClass` variable as you would be sure that it will always have a value. See below the diff below:
```diff
-  let buttonClass = 'button';
+  let buttonClass = `button ${color}`;
-
-  if (color) {
-    buttonClass += ` ${color}`;
-  }
```

What do you think? 🤔

Reply:
I'm doing that to avoid the extra space in case color is not defined, is not necessary, but I don't want the extra space

Reply:
I don't understand, what extra space?

Reply:
alright!